src/components/GestureComponents.tsx

```diff
@@ -63,24 +60,10 @@ export const FlatList = React.forwardRef<RNFlatList<any>, RNFlatListProps<any>>(
       renderScrollComponent={(scrollProps) => <ScrollView {...scrollProps} />}
     />
   )
-);
+) as <ItemT = any>(
+  props: PropsWithChildren<RNFlatListProps<ItemT>> &
+    RefAttributes<RNFlatList<ItemT>>,
+  ref: ForwardedRef<RNFlatList<ItemT>>
+) => ReactElement | null;
 // eslint-disable-next-line @typescript-eslint/no-redeclare
-export type FlatList<ItemT> = React.ComponentType<
-  RNFlatListProps<ItemT> &
-    NativeViewGestureHandlerProps &
```

Comment:
I removed `NativeViewGestureHandlerProps` from the type definition, because they were causing TypeScript errors. I don't think they were needed, since from what I can see they were not passed to the `ScrollView` component.

Reply:
Removing those will break users' code using things like `waitFor` or `simultanousHandlers`.